File: back-end/src/main/java/com/meutcc/backend/content/exercise/ExerciseResponseDTO.java
package com.meutcc.backend.content.exercise;

import com.meutcc.backend.content.exercise.dtos.ExerciseStatisticsDTO;
import com.meutcc.backend.content.question.QuestionDisplayMode;
import com.meutcc.backend.content.question.QuestionResponseDTO;

import java.time.LocalDateTime;
import java.util.List;

public record ExerciseResponseDTO(
        Long id,
        String title,
        String description,
        String instructions,
        Integer totalPoints,
        Integer passingScore,
        Integer timeLimit,
        Integer maxAttempts,
        Boolean shuffleQuestions,
        Boolean shuffleOptions,
        Boolean showCorrectAnswers,
        Boolean showScore,
        Boolean allowReview,
        QuestionDisplayMode questionDisplayMode,
        Boolean isActive,
        LocalDateTime availableFrom,
        LocalDateTime availableUntil,
        Boolean isAvailable,
        Integer questionsCount,
        List<QuestionResponseDTO> questions,
        ExerciseStatisticsDTO statistics,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {}

File: front-end/src/shared/types/ExerciseResponseDTO.ts
import type { ExerciseStatisticsDTO } from "@/shared/types/ExerciseStatisticsDTO";
import type { QuestionDisplayMode } from "@/shared/types/QuestionDisplayMode";
import type { QuestionResponseDTO } from "@/shared/types/QuestionResponseDTO";

export interface ExerciseResponseDTO {
  id: number;
  title: string;
  description?: string;
  instructions?: string;
  totalPoints: number;
  passingScore: number;
  timeLimit?: number;
  maxAttempts: number;
  shuffleQuestions: boolean;
  shuffleOptions: boolean;
  showCorrectAnswers: boolean;
  showScore: boolean;
  allowReview: boolean;
  questionDisplayMode: QuestionDisplayMode;
  isActive: boolean;
  availableFrom?: string;
  availableUntil?: string;
  isAvailable?: boolean;
  questionsCount?: number;
  questions?: QuestionResponseDTO[];
  statistics?: ExerciseStatisticsDTO;
  createdAt: string;
  updatedAt: string;
}